### PR TITLE
[Feature] support iceberg map struct

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [ ] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -353,11 +358,26 @@ public static Type convertColumnType(org.apache.iceberg.types.Type icebergType)
...
+                } else {
+                    return Type.UNKNOWN_TYPE;
+                }
+            case STRUCT:
+                List<Types.NestedField> fields = icebergType.asStructType().fields();
+                Preconditions.checkArgument(fields.size() > 0);
+                ArrayList<StructField> structFields = new ArrayList<>(fields.size());
+                for (Types.NestedField field : fields) {
+                    String fieldName = field.name();
+                    Type fieldType = convertColumnType(field.type());
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java:374]: Add a check, if fieldType is UNKNOWN_TYPE, we should return UNKOWN_TYPE directly.

Struct doesn't support unknown type's subfield.

**Review by Smith-Cruise — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java (hunk 1 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java:374]: GET

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java [hunk 2]
```diff
@@ -371,4 +391,14 @@ public static Database convertToSRDatabase(String dbName) {
...
     }
+
+    private static Type convertToMapType(org.apache.iceberg.types.Type icebergType) {
+        Type keyType = convertColumnType(icebergType.asMapType().keyType());
+        // iceberg support complex type as key type, but sr is not supported now
+        if (keyType.isComplexType()) {
+            return Type.UNKNOWN_TYPE;
+        }
+        return new MapType(convertColumnType(icebergType.asMapType().keyType()),
+                convertColumnType(icebergType.asMapType().valueType()));
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java:402]: be aware of unknown type

**Review by Smith-Cruise — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 23 (95.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[925]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|22|22|100.00%|[]|

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 23 (95.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[925]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|22|22|100.00%|[]|

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java [hunk 3]
```diff
@@ -353,11 +358,29 @@ public static Type convertColumnType(org.apache.iceberg.types.Type icebergType)
                 } else {
                     return Type.UNKNOWN_TYPE;
                 }
+            case MAP:
+                Type mapType = convertToMapType(icebergType);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergUtil.java:362]: Do we support nested map? like Map<Interger, Map<>>

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**